contracts/ERC4626/VenusERC4626Factory.sol

```diff
@@ -0,0 +1,112 @@
+    /// @param asset The base asset used by the vault
+    /// @param vault The vault that was created
+    event CreateERC4626(address indexed asset, ERC4626Upgradeable indexed vault);
+
+    /// @notice Emitted when the reward recipient address is updated.
+    /// @param oldRecipient The previous reward recipient address.
+    /// @param newRecipient The new reward recipient address.
+    event RewardRecipientUpdated(address indexed oldRecipient, address indexed newRecipient);
+
+    /// @notice Thrown when the provided comptroller is invalid (not registered in PoolRegistry)
+    error VenusERC4626Factory__InvalidComptroller();
+
+    /// @notice Thrown when trying to deploy a VenusERC4626 vault using an asset without a vToken
+    error VenusERC4626Factory__VTokenNonexistent();
+
+    /// @notice Initializes the contract
+    /// @param accessControlManager Address of the ACM contract
+    /// @param poolRegistryAddress Address of the Pool Registry contract
+    /// @param rewardRecipientAddress Reward recipient address
+    /// @param venusERC4626Implementation Address of the VenusERC4626 implementation contract
+    /// @param loopsLimitNumber The loops limit for the MaxLoopsLimit helper
+    function initialize(
+        address accessControlManager,
+        address poolRegistryAddress,
+        address rewardRecipientAddress,
+        address venusERC4626Implementation,
+        uint256 loopsLimitNumber
+    ) external initializer {
+        // checks are missing
+        __Ownable2Step_init();
+        __AccessControlled_init_unchained(accessControlManager);
+
+        poolRegistry = PoolRegistryInterface(poolRegistryAddress);
+        rewardRecipient = rewardRecipientAddress;
+        loopsLimit = loopsLimitNumber;
+
+        // Deploy the upgradeable beacon with the initial implementation
+        beacon = new UpgradeableBeacon(venusERC4626Implementation);
+    }
+
+    /// @notice Sets a new reward recipient address
+    /// @param newRecipient The address of the new reward recipient
+    /// @custom:access Requires access control
+    /// @custom:error ZeroAddressNotAllowed is thrown when the new recipient address is zero
+    /// @custom:event RewardRecipientUpdated is emitted when the reward recipient address is updated
+    function setRewardRecipient(address newRecipient) external {
+        _checkAccessAllowed("setRewardRecipient(address)");
+        ensureNonzeroAddress(newRecipient);
+
+        emit RewardRecipientUpdated(rewardRecipient, newRecipient);
+        rewardRecipient = newRecipient;
+    }
+
+    /// @notice Creates a VenusERC4626 vault for a given asset and comptroller
+    /// @param comptroller The comptroller's address to which asset belongs
+    /// @param asset The ERC20 asset address belongs to the comptroller
+    /// @return vault The deployed VenusERC4626 vault
+    function createERC4626(address comptroller, address asset) external returns (ERC4626Upgradeable vault) {
```

Comment:
Wouldn't it be more intuitive if we receive the address of a VToken here. This factory will deploy VenusERC4626 contracts, and each VenusERC4626 contract is associated with a VToken